MaterialLib/MPL/Properties/CapillaryPressureSaturation/CapillaryPressureVanGenuchten.h

```diff
@@ -0,0 +1,147 @@
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * Created on March 20, 2020, 9:59 AM
+ */
+
+#pragma once
+
+#include "SaturationVanGenuchten.h"
+
+namespace MaterialPropertyLib
+{
+ /// \copydoc MaterialPropertyLib::SaturationVanGenuchten
+class CapillaryPressureVanGenuchten : public SaturationVanGenuchten
+{
+public:
+    CapillaryPressureVanGenuchten(double const residual_liquid_saturation,
+                                  double const residual_gas_saturation,
+                                  double const exponent,
+                                  double const entry_pressure,
+                                  double const max_capillary_pressure)
+        : SaturationVanGenuchten(residual_liquid_saturation,
+                                 residual_gas_saturation, exponent,
+                                 entry_pressure),
+          _pc_max(max_capillary_pressure)
+    {
+    }
+
+    void setScale(std::variant<Medium*, Phase*, Component*> scale_pointer)
+    {
+        if (!std::holds_alternative<Medium*>(scale_pointer))
+        {
+            OGS_FATAL(
+                "The property 'CapillaryVanGenuchten' is implemented on the "
+                "'media' scale only.");
+        }
+        this->_medium = std::get<Medium*>(scale_pointer);
+    }
+
+    /// gets \f$ p_c(S) \f$.
+    PropertyDataType value(VariableArray const& variable_array,
+                           ParameterLib::SpatialPosition const& pos,
+                           double const t,
+                           double const dt) const;
+
+    /// gets \f$ \frac{\partial p_c(S)}{\partial  S} \f$
+    PropertyDataType dValue(VariableArray const& variable_array,
+                            Variable const variable,
+                            ParameterLib::SpatialPosition const& pos,
+                            double const t,
+                            double const dt) const;
+
+    /// gets \f$ \frac{\partial^2 p_c(S)}{\partial  S^2} \f$
+    PropertyDataType d2Value(VariableArray const& variable_array,
+                             Variable const variable1, Variable const variable2,
+                             ParameterLib::SpatialPosition const& pos,
+                             double const t, double const dt) const;
+
+protected:
+    double const _pc_max;  ///< Maximum capillary pressiure
+};
+
+/**
+ * \copydoc SaturationVanGenuchten
+ *
+ *  This class handles the computation of \f$ p_c(S) \f$ with the capillary
+ *  pressure regularization.
+ *
+ *   \brief The regularized van Genuchten model, please ref to
+ *   Marchand E, Mueller T, Knabner P.
+ *   Fully coupled generalized hybrid-mixed finite element approximation of
+ *   two-phase two-component flow in porous media.
+ *   Part I: formulation and properties of the mathematical model. Comput Geosci
+ *   2013;17(2):431-442.
+ */
+class RegularizedCapillaryVanGenuchten final
```

Comment:
only one property per translation unit.

Reply:
Changed accordingly.

Reply:
No, not changed yet. Two properties in same translation unit.

Reply:
Updated.